stage5.1_fuller_application/pycasa/model/image_folder.py

```diff
@@ -0,0 +1,48 @@
+# General imports
+import os
+
+import pandas as pd
+import numpy as np
+
+# ETS imports
+from traits.api import Directory, HasStrictTraits, Instance
+
+# Local imports
+from .image_file import ImageFile, SUPPORTED_FORMATS
+
+
+class ImageFolder(HasStrictTraits):
+    """ Model to hold an image folder.
+    """
+    path = Directory
+
+    data = Instance(pd.DataFrame)
+
+    def __init__(self, **traits):
+        # Don't forget this!
+        super(ImageFolder, self).__init__(**traits)
+        if not os.path.isdir(self.path):
+            msg = f"Unable to create an ImageFolder from {self.path} since" \
+                  f" it is not a valid directory."
+            raise ValueError(msg)
```

Comment:
Can this also be accomplished with `path = Directory(exists=True)`?

Reply:
That's true, but 1. I didn't know about it and 2. my goal was to introduce the `__init__` thing which is more fundamental. Maybe in the training, we can have a section about exploring Trait's keyword args, and show that off with this. Thanks @siddhantwahal !

Reply:
Ps First step is me replacing my models with yours anyway...

Reply:
> That's true, but 1. I didn't know about it and 2. my goal was to introduce the `__init__` thing which is more fundamental. Maybe in the training, we can have a section about exploring Trait's keyword args, and show that off with this and maybe one Editor keyword arg. In fact, I already have a slide for this! Thanks @siddhantwahal !

Sounds good! I actually didn't know about this, but there's a `trait_init` method which automatically calls `super().__init__`: https://docs.enthought.com/traits/traits_tutorial/introduction.html?highlight=initializer#the-traits-init-method